nimterop/build.nim

```diff
@@ -806,36 +802,59 @@ proc buildLibrary(lname, outdir, conFlags, cmakeFlags, makeFlags: string): strin
           gen = "Unix Makefiles".quoteShell
         if findExe("ccache").len != 0:
           gen &= " -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
-        makePath = outdir / "buildcache"
-        cmake(makePath, "Makefile", &".. -G {gen} {cmakeFlags}")
-        cmakeDeps = true
+        result.buildPath = outdir / "buildcache"
+        cmake(result.buildPath, "Makefile", &".. -G {gen} {flags}")
+        result.built = true
       else:
-        cmakeDepStr &= "cmake executable missing"
+        result.error = "cmake capable but cmake executable missing"
+  else:
+    result.buildPath = outdir
+
+proc buildWithConfMake(outdir, flags: string): BuildStatus =
+  if not fileExists(outdir / "Makefile"):
+    if findExe("bash").len != 0:
+      for file in @["configure", "configure.ac", "configure.in", "autogen.sh", "build/autogen.sh"]:
+        if fileExists(outdir / file):
+          configure(outdir, "Makefile", flags)
+          result.buildPath = outdir
+          result.built = true
+          break
+    else:
+      result.error = "configure capable but bash executable missing"
+  else:
+    result.buildPath = outdir
 
-    if not cmakeDeps:
-      if findExe("bash").len != 0:
-        for file in @["configure", "configure.ac", "configure.in", "autogen.sh", "build/autogen.sh"]:
-          if fileExists(outdir / file):
-            configure(outdir, "Makefile", conFlags)
-            conDeps = true
+proc buildLibrary(lname, outdir, conFlags, cmakeFlags, makeFlags: string, preferredMakeType: MakeType): string =
+  var
+    lpath = findFile(lname, outdir, regex = true)
+    makeFlagsProc = &"-j {getNumProcs()} {makeFlags}"
+    makePath = outdir
 
-            break
-      else:
-        conDepStr &= "bash executable missing"
+  if lpath.len != 0:
+    return lpath
 
-  if fileExists(makePath / "Makefile"):
-    make(makePath, lname, makeFlagsProc, regex = true)
-    made = true
+  var buildStatus: BuildStatus
 
-  var
-    error = ""
-  if not cmakeDeps and cmakeDepStr.len != 0:
-    error &= &"cmake capable but {cmakeDepStr}\n"
-  if not conDeps and conDepStr.len != 0:
-    error &= &"configure capable but {conDepStr}\n"
-  if error.len == 0:
-    error = "No build files found in " & outdir
-  doAssert cmakeDeps or conDeps or made, &"\n# Build configuration failed - {error}\n"
+  # Simply reverse order if we want configure/make vs CMake/make
+  case preferredMakeType
+  of mtCMake:
+    buildStatus = buildWithCmake(makePath, cmakeFlags)
+    if not buildStatus.built:
+      buildStatus = buildWithConfMake(makePath, conFlags)
+  of mtConfMake:
+    buildStatus = buildWithConfMake(makePath, conFlags)
+    if not buildStatus.built:
+      buildStatus = buildWithCmake(makePath, cmakeFlags)
+
+  if buildStatus.buildPath.len > 0:
+    buildStatus.built = findFile(lname, buildStatus.buildPath, regex = true).len > 0
+
+    if not buildStatus.built and fileExists(buildStatus.buildPath / "Makefile"):
```

Comment:
This means that if configure or cmake is run, make will not be called. But it needs to be called. See line 807.

Reply:
No, it actually will be called because of the line above. But it's probably clearer if I add a different variable here instead of reusing buildStatus.built